src/array.vala

```diff
@@ -77,14 +83,13 @@ public class Vast.Array : Object
                   size_t    origin  = 0)
         requires (scalar_size > 0)
         requires (shape.length > 0)
-        requires (_size_from_shape (shape) > 0)
```

Comment:
This is important no? It's prevents zeroes in shapes, which basically collapse the whole array.